src/code/z_actor.c

```diff
@@ -1744,11 +1744,11 @@ void func_8002F994(Actor* actor, s32 arg1) {
     actor->flags |= ACTOR_FLAG_28;
     actor->flags &= ~(ACTOR_FLAG_19 | ACTOR_FLAG_20 | ACTOR_FLAG_21);
     if (arg1 < 40) {
-        actor->sfx = NA_SE_PL_WALK_DIRT - SFX_FLAG;
+        actor->sfx = 3;
     } else if (arg1 < 100) {
-        actor->sfx = NA_SE_PL_WALK_CONCRETE - SFX_FLAG;
+        actor->sfx = 2;
     } else {
-        actor->sfx = NA_SE_PL_WALK_SAND - SFX_FLAG;
+        actor->sfx = 1;
```

Comment:
Can you elaborate on this change? This doesn't play a sfx?

Reply:
No, this is confusing to explain, but these values are actually timer states that get fed as ioData into ioPort, that then triggers seq 0 to change the tempo of the ticking timer. This is all programmed internally in seq 0

Reply:
https://github.com/engineer124/oot/blob/75ea3511865b441471945afeded76af1501a01d7/src/code/z_actor.c#L2280

Reply:
I see (more or less)
Could you put a comment until this is documented, maybe something like `With ACTOR_FLAG_28 the sfx value is not a sfx id, see func_80030ED8`

Reply:
Even more specifically, they're indices to this table here, which then applies a delay to when the next "tick" is in the timer 
https://github.com/MNGoldenEagle/oot/blob/fb5bcb9e5e9bfa5e9735c9a3015aa32d896a1767/assets/sequences/sfxbanks/env.seq.inc#L740-L744
Except the indices are offset by 1 (`actor->sfx = 1; // index 0`) so that it passes the `actor->sfx != 0` check needed to play the sfx. Then the `(s8)(actor->sfx - 1)` gets the true index

Yeah, I can add a temporary comment. I'll also be tackling this function in the near future as well